blues/sparrow-reference-web-app
# bug: fix event ingestion and remove bulk import functionality

# Problem Context

A user ran into a problem when attempting to import events in the Sparrow Reference Web App [Routing data from NoteHub to Sparrow reference web app](https://discuss.blues.io/t/routing-data-from-notehub-to-sparrow-reference-web-app/1310).

I was able to recreate the issue locally. The main problem is that the app’s bulk import process and routed event ingestion handler were never updated to handle the updated event structure from Notehub, and newer events fail to parse with a “device uid is not defined” error.

Recommended fix is remove bulk import and update the event ingestion method. We can update the documentation to discuss Notehub’s re-routing ability [Notehub Walkthrough - Blues Developers](https://dev.blues.io/notehub/notehub-walkthrough/#manually-routing-events) instead of bulk import.

## Changes

* Updated the shape of routed Notehub events to reflect the new structure, and fix the code that was throwing errors and preventing the data from being stored in the Postgres DB
* Remove all traces of bulk import functionality - it's no longer needed now that Notehub offers event re-routing
* Update docs and a few error messages to make users aware of Notehub re-routing functionality for their events

## Screenshot (if applicable)

![Screenshot 2023-04-05 at 9 46 57 AM](https://user-images.githubusercontent.com/20400845/230099947-e40629b9-4880-43d7-9f9c-e386c2846520.png)

## Testing

Unit / integration / e2e tests?

 All tests pass, code compiles.

Steps to test manually?

1. Set up Postgres DB, set up localtunnel or ngrok connection to ingest events from Notehub route, start DB and app running
2. Route event from Notehub to app manually
3. See event get successfully handled by app, and event data show up on screen in appropriate node or gateway.

Any other sorts of testing notes to include?

## Any other related PRs

n/a

## Ticket(s)

https://trello.com/c/MaIZSt9d/82-sparrow-reference-web-app-bulk-import-event-processing-fails
